www/js/app.js

```diff
@@ -813,8 +825,16 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
 
             // Convert links into javascript calls
             $('#articleContent').contents().find('body').find('a').each(function() {
-                // Store current link's url
-                var url = $(this).attr("href");
+                // Compute current link's url (with its namespace), if applicable
+                var fullUrl = this.href;
+                var urlMatch = fullUrl.match(regexpUrl);
+                var url;
+                if (urlMatch) {
+                    url = urlMatch[0];
+                }
+                else {
+                    url = $(this).attr("href");
+                }
```

Comment:
@mossroy I couldn't resist trying all this as a one-liner....!
`var url = regexpUrl.test(this.href) ? this.href.match(regexpUrl)[0] : $(this).attr("href");`
I'm half joking, but perhaps there's no need to define `fullUrl`, when you could just match against `this.href`.